### PR TITLE
[Issue 2554] clean up fetcher frontend data fetcher pattern

### LINKED ISSUE #2554: Cleanup Fetcher Pattern
### Summary

The current search fetch pattern was introduced over the course of two PRs:
* https://github.com/HHS/simpler-grants-gov/pull/1263
* https://github.com/HHS/simpler-grants-gov/pull/1386

Some functionality was created here to abstract away the actual classes that are being used to make the requests, with the benefit being the ability switching out real implementations for mocks when in testing scenarios based on an environment variable. However, it seems like this environment variable (`USE_SEARCH_MOCK_DATA`) is not in use anywhere, and mocking of API interactions is being done in other ways in various tests at the moment.

Since the functionality this abstraction offers is not being used, and there are arguably better alternatives in use, I propose removing the "fetcher" pattern present in the services directory and allow components that make requests to the API to directly access the relevant API classes. This should improve readability and maintainability of the code, and create a clearer pattern to use going forward as we add more API related functionality.

### Acceptance criteria

* All calls to search and opportunity endpoints function as expected
* Components making calls to the API use the BaseAPI extensions directly rather than "fetchers"
* Functionality present in the files at src/services/ other than the FeatureFlagManager and SearchFilterManager are either removed or moved into the BaseAPI or extensions

### Alternative

We opt to maintain the fetcher pattern and update our testing practices and API interactions to use it

### PR DESCRIPTION
**Pull request by doug-s-nava**
## Summary
Fixes #2554

## Changes proposed

This change removes the `services/search/searchfetcher` directory, and moves any necessary related functionality into the `/api` classes. For more discussion on reasoning and expectations, refer to the ticket

Note that for ease of code change management this is currently branched off of [dschrashun/2448-v1-search-endpoint
](https://github.com/HHS/simpler-grants-gov/pull/2518) which has not been merged to main yet. If this PR is approved before feature/frontend-opensearch is approved for merging to main, this will merge into freature/frontend-opensearch. This seems fine as the change isn't urgent but could be rebased off main if we want to get it in.

### Test steps

1. _VERIFY_ all search operations work as expected
2. _VERIFY_ opportunity detail pages load as expected

## Additional information
 Follow up todos:

- [ ] refactor opportunity detail page to properly use suspense skeletons
- [ ] figure out a way to properly mock API interactions and allow for page level testing without mocking suspense (try upgrading to next 15 / react 19?)

### CODE REVIEW

File: frontend/src/app/api/Fetchers.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import SearchOpportunityAPI from "src/app/api/SearchOpportunityAPI";
+
+import OpportunityListingAPI from "./OpportunityListingAPI";
```
Comment (emilycnava) [inline frontend/src/app/api/Fetchers.ts:3]: Maybe I"m just not seeing it, but not seeing this file present in this PR?

**Review by emilycnava — COMMENTED** [review]
Haven't run locally yet, but scanned over code and it mostly tracks, albeit it looks like you're referencing a file that's not defined?

File: frontend/src/app/api/Fetchers.ts (hunk 1 above)
Comment (doug-s-nava) [reply frontend/src/app/api/Fetchers.ts:3]: that file should be there, but the relative path needs to get cleaned up. You don't see /simpler-grants-gov/frontend/src/app/api/OpportunityListingAPI.ts? It should be there on main as well

File: frontend/src/services/search/searchfetcher/APISearchFetcher.ts
Comment (acouch) [inline frontend/src/services/search/searchfetcher/APISearchFetcher.ts:1]: 🔥

File: frontend/src/services/search/searchfetcher/MockSearchFetcher.ts
Comment (acouch) [inline frontend/src/services/search/searchfetcher/MockSearchFetcher.ts:1]: 🔥

File: frontend/src/services/search/searchfetcher/SearchFetcher.ts
Comment (acouch) [inline frontend/src/services/search/searchfetcher/SearchFetcher.ts:1]: 🔥

File: frontend/src/services/search/searchfetcher/SearchFetcherUtil.ts
Comment (acouch) [inline frontend/src/services/search/searchfetcher/SearchFetcherUtil.ts:1]: 🔥

**Review by acouch — APPROVED** [review]
Huge improvement!

[nit]: The `src/services/search/searchfetcher` folder has only one file in it which could be moved to the `src/services/search` folder.

**Pull request merged**